extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/ColorSplasher.pushbutton/script.py

```diff
@@ -164,133 +142,136 @@ def Execute(self, uiapp):
+                        filters = view.GetFilters()
+                        if len(filters) != 0:
+                            for filt_id in filters:
+                                filt_ele = new_doc.GetElement(filt_id)
+                                if filt_ele.Name.StartsWith(filter_name):
+                                    view.RemoveFilter(filt_id)
+                                    try:
+                                        new_doc.Delete(filt_id)
+                                    except:
+                                        pass
+                    except Exception as e:
+                        pass
+                    # Reset visibility
+                    for id in collector:
+                        view.SetElementOverrides(id, ogs)
         except Exception as e:
             pass
             
     def GetName(self):
         return "Reset colors in elements"
 
-class createLegend(IExternalEventHandler):
+
+class createLegend(UI.IExternalEventHandler):
     def __init__(self):
         pass
         
     def Execute(self, uiapp):
         try:
             new_doc = uiapp.ActiveUIDocument.Document
-            #Get legend view
-            collector = FilteredElementCollector(new_doc).OfClass(Autodesk.Revit.DB.View).ToElements()
-            legends=[]
+            rvt_ver = int(uiapp.Application.VersionNumber)
+            # Get legend view
+            collector = DB.FilteredElementCollector(new_doc).OfClass(DB.View).ToElements()
+            legends = []
             for vw in collector:
-                if vw.ViewType == ViewType.Legend:
+                if vw.ViewType == DB.ViewType.Legend:
                     legends.Add(vw)
                     break
-            if len(legends)>0:
-                #Duplicate existing legend
-                t = Transaction(new_doc, "Create Legend")
+            if len(legends) > 0:
+                # Duplicate existing legend
+                t = DB.Transaction(new_doc, "Create Legend")
                 t.Start()
-                trans = SubTransaction(new_doc)
+                trans = DB.SubTransaction(new_doc)
                 trans.Start()
-                new_id_legend = legends[0].Duplicate(ViewDuplicateOption.Duplicate)
+                new_id_legend = legends[0].Duplicate(DB.ViewDuplicateOption.Duplicate)
                 newLegend = new_doc.GetElement(new_id_legend)
-                i=1
+                i = 1
                 while True:
                     try:
-                        newLegend.Name = "Color Legend " + str(i)
+                        if rvt_ver < 2025:
+                            newLegend.Name = unicode("Color Legend " + str(i))
+                        else:
+                            newLegend.Name = "Color Legend " + str(i)
                         break
                     except:
-                        i+=1
+                        i += 1
                         if i == 1000:
```

Comment:
Turn it into a `for i in range(1000)` loop, we break out of it once a good name is found anyway